### PR TITLE
Issue1138 by Pjohans: hide pay-selected button when loan provider is …

### PR DESCRIPTION
**Pull request by pjohans**
…integra - it only supports payment off ALL debts in one go

### CODE REVIEW

File: plugins/content_types/debts/debts.inc [hunk 1]
```diff
@@ -76,15 +76,19 @@ function ding_debt_debts_form($form, &$form_state, $debts, $has_invoiced_fees, $
...
     if (!empty($provider)) {
-      $form['pay_selected'] = array(
-        '#type' => 'submit',
-        '#prefix' => '<div class="pay-button">',
-        '#value' => t('Pay selected debts'),
-        '#suffix' => '</div>',
-        '#submit' => array('ding_debt_debts_form_submit_pay_selected'),
-      );
+      // Integra only support payment of ALL debts in one go.
+      if ($loan_provider != 'openruth') {
```
Comment (cableman) [inline plugins/content_types/debts/debts.inc:83]: This should be done inside the provider and not in the debt module. The objective with the provider layer is that the system is independent of the individual providers.

So the right approach would be to ask the provider layer: Do the currently active provider supports payment of selected fees.

**Comment by pjohans** [comment]
That would take extra levels in the provider to describe various scenarios. like :
'debt' => array(
        'prefix' => 'debt',
              'transaction' => 'all'
        'file' => $path . '/includes/alma.debt.inc',
      ),
As it is now the provider only says if it supports or not, and the fact that integra only handles FULL patyment is implemented in openruth::openruth_debt_payment_received.

**Comment by kasperg** [comment]
I agree with @cableman that this is not the right way to go about this.

@pjohans is right that [openruth already has some handling of this](https://github.com/ding2/openruth/blob/master/includes/openruth.debt.inc#L90-L101) but the original issue shows that the amount is still withdrawn from the credit card of the client which is a major issues as it is. The fact that the the debt is not marked as payed only means that the system errs in advantage of itself - not the user.

It would be preferable if we could cancel the payment at this point but I do not think that is possible.

I also agree that adding new provider calls to check for payment support could be excessive - especially given the expected short lifetime of Integra.

I suggest that the `openruth` module implements a `hook_ding_debt_debts_form_form_alter()` and removed the button in question.

**Comment by pjohans** [comment]
okay

**Comment by kasperg** [comment]
Closing this in favor of ding2/openruth#12.

**Pull request closed without merging**